pkg/endpoint/bpf.go

```diff
@@ -870,6 +870,11 @@ func (e *Endpoint) runPreCompilationSteps(regenContext *regenerationContext) (he
 		if err = e.writeHeaderfile(nextDir); err != nil {
 			return false, fmt.Errorf("unable to write header file: %s", err)
 		}
+	} else if datapathRegenCtxt.regenerationLevel == regeneration.RegenerateWithDatapathRebuild {
```

Comment:
We could probably tweak this to avoid duplicating the `writeHeaderfile()` call by instead having the `headerfileChanged` check above only set the `regenerationLevel` then have this check do the following instead:

```suggestion
	if datapathRegenCtxt.regenerationLevel >= regeneration.RegenerateWithDatapathRewrite {
```

Then I'm not sure whether the comment below provides any additional context; basically if either a rewrite or rebuild is required, we'll be sure to write the headerfile again.

Reply:
Done